src/test/java/fi/helsinki/cs/tmc/core/commands/SubmitTest.java

```diff
@@ -45,31 +49,75 @@
 import java.text.ParseException;
 import java.util.ArrayList;
 import java.util.List;
-import static org.mockito.Matchers.eq;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
 
 public class SubmitTest {
 
+
     private Submit submit;
     private ExerciseSubmitter submitterMock;
     private CoreTestSettings settings;
     private String submissionUrl;
     private ProgressObserver observer;
 
-    @Rule public WireMockRule wireMock = new WireMockRule();
+    @Rule public WireMockRule wireMock = new WireMockRule(0);
+    private String serverAddress = "http://127.0.0.1:";
 
     private Submit submitWithObserver;
     private SubmissionPoller pollerMock;
     private String path;
 
-    @Before
-    public void setup() throws Exception {
+    private void createSettings() {
         settings = new CoreTestSettings();
         settings.setUsername("Samu");
         settings.setPassword("Bossman");
         settings.setCurrentCourse(new Course());
         settings.setApiVersion("7");
+    }
+
+    private void buildWireMock() throws URISyntaxException {
+        wireMock.stubFor(
+                post(urlPathEqualTo("/exercises/1231/submissions.json"))
+                        .willReturn(
+                                WireMock.aResponse()
+                                        .withStatus(200)
+                                        .withBody(
+                                                ExampleJson.failedSubmitResponse.replace(
+                                                        "https://tmc.mooc.fi/staging",
+                                                        serverAddress)
+                                                        .replaceAll(
+                                                                "8080",
```

Comment:
I'm wondering whether we might accidentally replace more than the port... maybe like {{PORT}} or smthing...
I guess that it is ok for now like this.